Document new choice of writing style guide

activus-d opened the pull request:
This pull request resolves [GSoD: Google developer documentation style guide and American English for our docs](https://github.com/wagtail/wagtail/issues/10608)

lb- reviewed (commented):
Can we cross-reference this update here in our other general guidelines page?

* `docs/contributing/general_guidelines.md`
* Maybe we should move the Latin phrases thing on that page into these documentation guidelines?
* In the **language** section we should clarify that the documentation language is different with a link to this documentation guideline section. Similar to how we call out that "identifiers within code should use American English...".

---

Currently that page has this content at the start for reference

```md
# General coding guidelines

## Language

British English is preferred for user-facing text; this text should also be marked for translation (using the `django.utils.translation.gettext` function and `{% translate %}` template tag, for example). However, identifiers within code should use American English if the British or international spelling would conflict with built-in language keywords; for example, CSS code should consistently use the spelling `color` to avoid inconsistencies like `background-color: $colour-red`.

### Latin phrases and abbreviations

Try to avoid Latin phrases (such as `ergo` or `de facto`) and abbreviations (such as `i.e.` or `e.g.`), and use common English phrases instead. Alternatively, find a simpler way to communicate the concept or idea to the reader. The exception is `etc.` which can be used when space is limited.

Examples:

| Don't use this | Use this instead     |
| -------------- | -------------------- |
| e.g.           | for example, such as |
| i.e.           | that is              |
| viz.           | namely               |
| ergo           | therefore            |
```

thibaudcolas reviewed (changes requested):
I like this but it’s way too much @activus-d!

lb- commented on docs/contributing/documentation_guidelines.md:
````diff
@@ -13,6 +13,66 @@ Wagtail’s documentation uses a mixture of [Markdown](https://myst-parser.readt
+
+[*]
+BasedOnStyles = Vale, Google
+```
````
A bit of feedback on using this, I gave this a go with my PR https://github.com/wagtail/wagtail/pull/10590

- It seems to check inline code snippets e.g. `<a href=""...></a>` was checked for things like brackets which is incorrect, same with code blocks.
- It flags the myst internal refs as using too many brackets e.g. `(some_ref)=`, which it should ignore.
- It says `admin` should be `administrator` which would flag a lot of false errors in our existing documentation. Similarly it flags `application` should be `app` which may not be suitable.
- This base config misses a lot of spelling such as `Stimulus` and `Draftail` as proper nouns.
- Re-writing a lot of things into active (not passive) voice is hard, probably just my bad writing skills, I envisage many others will struggle with this.
- It did not seem to flag any issues with UK spelling (e.g. using `behaviour` not `behavior`), not sure if that's something to consider if we want to also adopt US spelling.

I think the vale checks are great and will keep trying it out but it may need some more work for us to have a baseline reliable config that's a good starting point for the majority of our documentation.

lb- replied:
One more note, not sure if it's my setup but after using Vale with VScode my mac slowed down a lot.

It appears to take up a large amount of memory.

<img width="679" alt="Screenshot 2023-07-17 at 8 29 26 am" src="https://github.com/wagtail/wagtail/assets/1396140/245f370c-a805-497c-a39e-261abbadc2b0">

activus-d commented:
@lb- @thibaudcolas I've added the instructions on how to use Vale as a comment in #10608. I also modified it slightly to include guidance on the proper use of Vale. This modification addressed some of the issues brought up by @lb- in https://github.com/wagtail/wagtail/pull/10634#discussion_r1264335385

thibaudcolas reviewed (approved):
Excellent! Sorry it took me forever to review this @activus-d.
